status/reporters.go

```diff
@@ -37,44 +38,48 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// GitHubReporter reports status back to GitHub for a Snapshot.
-type GitHubReporter struct {
-	logger    logr.Logger
-	k8sClient client.Client
-	client    github.ClientInterface
+// StatusUpdater is common interface used by status reporter to update PR status
+type StatusUpdater interface {
+	// Authentication of client
+	Authenticate(ctx context.Context, snapshot *applicationapiv1alpha1.Snapshot) error
+	// Update status of PR
+	UpdateStatus(ctx context.Context, integrationTestStatusDetail intgteststat.IntegrationTestStatusDetail) error
 }
 
-// GitHubReporterOption is used to extend GitHubReporter with optional parameters.
-type GitHubReporterOption = func(r *GitHubReporter)
-
-func WithGitHubClient(client github.ClientInterface) GitHubReporterOption {
-	return func(r *GitHubReporter) {
-		r.client = client
-	}
+// CheckRunStatusUpdater updates PR status using CheckRuns (when application integration is enabled in repo)
+type CheckRunStatusUpdater struct {
+	ghClient          github.ClientInterface
+	k8sClient         client.Client
+	logger            *helpers.IntegrationLogger
+	owner             string
+	repo              string
+	sha               string
+	snapshot          *applicationapiv1alpha1.Snapshot
+	creds             *appCredentials
+	allCheckRunsCache []*ghapi.CheckRun
 }
 
-// NewGitHubReporter returns a struct implementing the Reporter interface for GitHub
-func NewGitHubReporter(logger logr.Logger, k8sClient client.Client, opts ...GitHubReporterOption) *GitHubReporter {
-	reporter := GitHubReporter{
-		logger:    logger,
+func NewCheckRunStatusUpdater(
```

Comment:
nit: Do we want to add some description for this func like:
Returns checkrunStatusUpdater for updating PR status.

Reply:
added